src/components/apphost.js

```diff
@@ -308,12 +308,17 @@ function askForExit() {
         exitPromise = actionsheet.show({
             title: globalize.translate('MessageConfirmAppExit'),
             items: [
+                { id: 'logout', name: globalize.translate('ButtonSignOut') },
```

Comment:
> Shouldn't logout menu entry be added only if the user is logged in?

Good point, I should add a check for that.


> Also, I don't think logout should be the default action.

That could be a point for discussion indeed. IMO not many people would actually go through this menu to close the app on their TV, if the app even is fully closed at all. More users would probably be switching between accounts.
Seeing as putting this option between 'Yes' and 'No' does not feel right aesthetically, and way at the bottom messes with the UX of having cancel / ignore as the always last option this seemed to be the best place to keep it for now.

Reply:
> IMO not many people would actually go through this menu to close the app on their TV, if the app even is fully closed at all.

This menu is what Samsung suggests for exiting the app on `Back` action. And I think it does close it compared to opening another app - it doesn't restore the state when re-launching.

> Seeing as putting this option between 'Yes' and 'No' does not feel right aesthetically, and way at the bottom messes with the UX of having cancel / ignore as the always last option this seemed to be the best place to keep it for now.

In fact, `No` should be the default.
It can be marked as selected: https://github.com/jellyfin/jellyfin-web/blob/218db7a7c3f9cdd0f7ba980de4bff9bdb86946e3/src/components/actionSheet/actionSheet.ts#L20

Reply:
> It can be marked as selected:

I've been testing with that, but it seems this option only puts a checkmark in front of the item. Not actually highlighting it as the active selection.

Reply:
> would be likely to force-close it

Force-close is the last option to exit, not the correct way to close the app.

> Not actually highlighting it as the active selection.

In Chrome (device mode to emulate Tizen + a hack to call the menu) it is highlighted, but the checkmark is not desirable. We need to add an option that allows us to set the current menu item.

Reply:
> Force-close is the last option to exit, not the correct way to close the app.

Based on my experience with end users they don't care about the "correct way", but rather most convenient / easiest / or fastest way. Though that's a bit beside the point, it does illustrate why I believe the exit menu would be a good place to add a logout option. It's much more convenient than navigating through the menu to log out, and it's a better option to force closing the app to switch accounts.

I'm fine with discussing placement / sorting of the options in the exit menu and which should be highlighted by default. Though if changes need to be made to the way these menus are currently implemented that would best be handled in another issue.